packages/mobile/android/app/src/main/cpp/own-native-lib.cpp

```diff
@@ -141,17 +141,33 @@ int start_redirecting_stdout_stderr()
 // node's libUV requires all arguments being on contiguous memory.
 extern "C" jobject JNICALL
 Java_com_quietmobile_Backend_BackendWorker_startNodeWithArguments(
-    JNIEnv *env,
-    jobject /* this */,
-    jobjectArray arguments,
-    jstring modulesPath)
+        JNIEnv *env,
+        jobject /* this */,
+        jobjectArray arguments,
+        jstring modulesPath,
+        jstring dataPath)
 {
 
     // Set the builtin_modules path to NODE_PATH.
     const char *path_path = env->GetStringUTFChars(modulesPath, 0);
+    const char *logs = "/logs";
+    jstring logs_js = env->NewStringUTF(logs);
+    jclass cls_StringBuilder = env->FindClass("java/lang/StringBuilder");
+    jmethodID ctr_StringBuilder = env->GetMethodID(cls_StringBuilder, "<init>", "(I)V");
+    jobject stringBuilder = env->NewObject(cls_StringBuilder, ctr_StringBuilder, 100);
+    jmethodID mid_StringBuilder_append = env->GetMethodID(cls_StringBuilder, "append", "(Ljava/lang/String;)Ljava/lang/StringBuilder;");
+    for (auto str: { dataPath, logs_js }) {
+        env->CallObjectMethod(stringBuilder, mid_StringBuilder_append, str);
+    }
+    jmethodID mid_StringBuilder_toString = env->GetMethodID(cls_StringBuilder, "toString", "()Ljava/lang/String;");
+    auto js_log_path = (jstring) env->CallObjectMethod(stringBuilder, mid_StringBuilder_toString);
+    const char* log_path = env->GetStringUTFChars(js_log_path, nullptr);
```

Comment:
I mentioned before I'm not great with C++, this is what came up when I searched how to concatenate two `jstring` variables.  I have to imagine there's a simpler way but this works.